### PR TITLE
[Mellanox] Improve FW upgrade logging: add syslog logger

### PR DESCRIPTION
**Pull request by nazariig**
Signed-off-by: Nazarii Hnydyn <nazariig@nvidia.com>

Basically mlnx-fw-upgrade.sh is used in two places:
1. https://github.com/Azure/sonic-buildimage/blob/201811/files/scripts/syncd.sh#L109
```bash
        /usr/bin/mst start
        /usr/bin/mlnx-fw-upgrade.sh
        /etc/init.d/sxdkernel start
        /sbin/modprobe i2c-dev
```
2. https://github.com/Azure/sonic-buildimage/blob/201811/device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot#L32
```bash
ParseArguments "$@"

${FW_UPGRADE_SCRIPT} --upgrade --verbose
EXIT_CODE="$?"
```

In first case the `stdout` is redirected to `syslog` directly by `systemd`.
Thus, the `syslog` logger is only required in second case.

#### Why I did it
* To improve ASIC/CPLD FW upgrade logging
* To improve CPLD upgrade time

#### How I did it
* Added `syslog` logger support
* Replaced `_pciconf0` -> `_pci_cr0` to reduce CPLD upgrade time

#### How to verify it
1. mlnx-fw-upgrade.sh --upgrade

#### Which release branch to backport (provide reason below if selected)

- [x] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

#### Description for the changelog

* N/A

#### A picture of a cute animal (not mandatory but encouraged)
```
      .---.        .-----------
     /     \  __  /    ------
    / /     \(  )/    -----
   //////   ' \/ `   ---
  //// / // :    : ---
 // /   /  /`    '--
//          //..\\
       ====UU====UU====
           '//||\\`
             ''``
```

### CODE REVIEW

**Comment by nazariig** [comment]
@madhanmellanox FYI

File: platform/mellanox/mlnx-fw-upgrade.j2 [hunk 1]
```diff
@@ -219,7 +241,11 @@ function UpgradeASICFW() {
 function UpgradeCPLDFW_Worker() {
     local -r _CPLD_BURN_FILE="${1}"
     local -r _CPLD_REFRESH_FILE="${2}"
-    local -r _ASIC_DEV="$(find /dev/mst -iname '*_pciconf0')"
+    local -r _ASIC_DEV="$(find /dev/mst -iname '*_pci_cr0')"
```
Comment (qiluo-msft) [inline platform/mellanox/mlnx-fw-upgrade.j2:244]: >_pci_cr0 [](start = 49, length = 8)

Could you explain the motivation of this change? Is it a bug fix?

Comment (nazariig) [reply platform/mellanox/mlnx-fw-upgrade.j2:244]: @qiluo-msft done - updated the description

**Review by qiluo-msft — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Review by sujinmkang — APPROVED** [review]

**Pull request merged**